README.md

```diff
@@ -35,19 +35,21 @@ objectScan(['a.*.f'])({ a: { b: { c: 'd' }, e: { f: 'g' } } });
 - Partial key and index wildcard matching, e.g. `mark*` or `[1*]`
 - Infinite nested matches with `**`
 - Simple or-clause for key and index with `{a,b}` and `[{0,1}]`
+- Input is traversed exactly once and only unique results are returned.
 - Full support for escaping
 - Lots of tests to ensure correctness
 
 ### Options
 
-**Note on Functions:**
-Signature for all functions is `Fn(key, value, { parents, isMatch, needle, needles })`, where:
+**Note on Functions:** Signature for all functions is `Fn(key, value, { parents, isMatch, matches, needles })`, where:
 - `key` is the key that the function is called for (respects `joined` option).
 - `value` is the value of that key.
 - `parents` is an array containing all parents as `[parent, grandparent, ...]`. Contains parents that are arrays only iff `useArraySelector` is true.
-- `isMatch` is true if this is a valid (intermittent) result.
-- `needle` is the needle that matches if `isMatch` is true, otherwise `null`.
-- `needles` are all needles that triggered the function call.
+- `isMatch` is true if exactly matched by at least one key. Indicates valid (intermittent) result.
+- `matches` are all needles matching key exactly.
+- `needles` are all needles matching key (partially).
+
+Note: `matches` and `needles` can be hard to interpret when `**` is used.
```

Comment:
I've opened an issue for this [here](https://github.com/blackflux/object-scan/issues/158). But it was already a problem before and just became more evident with this pr. Will need to consider how to best fix it. Basically the problem is that '**' injections are not displayed in needles, since they are pre-computed and '**' is dynamically used based on input data.

Reply:
Incorrect. This is in fact a bug. Will fix and need to remove this comment.

Reply:
removed note